### ISSUE TITLE
[DOC] Use double backticks in python docstrings instead of a single one to improve appearance on Sphinx documentation

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#4524]
#### Describe the issue linked to the documentation

In a lot of docstrings, references to other variables, functions or methods are mentioned. Majority of them use double backticks (``), but others do not (especially a lot in extension templates). Using double backticks improve the appearance in the rendered website, as `reStructuredText` treat them [differently](https://docutils.sourceforge.io/docs/user/rst/quickref.html#inline-markup).

#### Suggest a potential alternative/fix

1. Find all occurrences of single backticks, for example searching for ``\s`[^`]`` regex pattern in VSCode may be a good point to start.
2. Replace these with double backtick.
3. Check updated documentation site as shown in PR link to see effects.

**Comment by fkiraly** [comment]
oh, we should be using double backticks in docstrings? Sorry for doing this wrong.

**Comment by yarnabrina** [comment]
I'm sorry if I explained insufficiently. It's not really wrong, I just felt it may look better.

Trying to show the difference with some examples below.

## double backticks (``` `` ```) around `feature_importances_` and `coef_`

![image](https://user-images.githubusercontent.com/39331844/235340091-dbb0b78f-3f12-47a8-9509-a17a784468b9.png)

![image](https://user-images.githubusercontent.com/39331844/235340111-8d5a5ade-10e7-4a2e-86ff-25ea66d6e7cc.png)

## single backtick (`` ` ``) around `n`, `forecasters` and `aggfunc`

![image](https://user-images.githubusercontent.com/39331844/235340186-6a62d712-0921-494e-b28e-9a7a766bcdb5.png)

![image](https://user-images.githubusercontent.com/39331844/235340198-c6caf16d-6947-4583-9cc4-cf40ce908932.png)

**Comment by fkiraly** [comment]
Yes, I see - we want as code blocks, not cursive.
I've been doing it wrong systematically.

Is there an error robust way to do this automatially?

**Comment by yarnabrina** [comment]
> Is there an error robust way to do this automatially?

If you mean if there is a way to make single backticks to be displayed in similar way as double backticks, I do not know. There are lots of Sphinx configurations, one of them, say [`default_role`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html) may be used to do some handling, may be as `any`, but I never tried.

On the other hand, one can do find-replace a bit catiously to modify existing occurrences.

**Comment by fkiraly** [comment]
> If you mean if there is a way to make single backticks to be displayed in similar way as double backticks, I do not know.

No, I meant, fixing this throughout the code base - replacing single backtick occurrences with double.

The easiest way that is safe I can think of would be a regex replace, where one needs to be cautious and have a pattern like space-tick-star-tick-space or similar, to avoid replacing the wrong ticks.

**Comment by yarnabrina** [comment]
> * Find all occurrences of single backticks, for example searching for `` \s`[^`] `` regex pattern in VSCode may be a good point to start.
> 
> * Replace these with double backtick.

I was thinking along these lines. Tried locally with the following, but it does not cover all cases, and cover some wrong cases. 

```
find=(?<=\s`)([^`]+)(?=`\s)
replace=`$1`
```

There are changes in more than 3000 files with above, so gave up for now as manual validation seemed too much.

**Comment by fkiraly** [comment]
Hm, we could try to find backticks *in docstrings*?

These are precisely the backticks that satisfy """ - wildcard - backtick - wildcard - """
